[CORE-213] - added language picker directive to main templates and ad…

TolgaCura opened the pull request:
…ded simple styling to picker

TolgaCura commented:
I have created a test branch for this in the demo-widget project, please see branch CORE-213 on gitlab

TolgaCura commented:
![screen shot 2016-12-15 at 11 20 01](https://cloud.githubusercontent.com/assets/19417743/21222291/0cd40a2e-c2b9-11e6-846e-69ac9f2ec442.png)
![screen shot 2016-12-15 at 11 20 33](https://cloud.githubusercontent.com/assets/19417743/21222290/0cd01018-c2b9-11e6-9851-1309be7b719c.png)
![screen shot 2016-12-15 at 11 21 19](https://cloud.githubusercontent.com/assets/19417743/21222293/0cd59240-c2b9-11e6-9511-532dbb6d1c97.png)
![screen shot 2016-12-15 at 11 21 38](https://cloud.githubusercontent.com/assets/19417743/21222292/0cd56838-c2b9-11e6-8be9-cb227e0105ec.png)
![screen shot 2016-12-15 at 11 23 10](https://cloud.githubusercontent.com/assets/19417743/21222295/0cd827f8-c2b9-11e6-81c4-aa7818d228cd.png)
![screen shot 2016-12-15 at 11 23 30](https://cloud.githubusercontent.com/assets/19417743/21222294/0cd763ae-c2b9-11e6-8b7f-f78caec78980.png)
![screen shot 2016-12-15 at 11 23 53](https://cloud.githubusercontent.com/assets/19417743/21222296/0ce38314-c2b9-11e6-8faa-f4fa9a2a962f.png)

lukeferguson commented on src/admin-booking/templates/main.html:
```diff
@@ -2,6 +2,8 @@
 
   <div bb-form-data-store></div>
 
+  <div bb-language-picker></div>
```
is the language picker hidden when there's only one available language?

TolgaCura replied:
the directive for the language picker handles showing/hiding the component
should this logic be implemented on the view?

see line 17: https://github.com/BookingBug/bookingbug-angular/blame/master/src/core/javascripts/i18n/directives/language_picker.directive.js.coffee

lukeferguson commented on src/core/templates/i18n/language_picker.html:
```diff
@@ -1,7 +1,10 @@
\ No newline at end of file
+<div id="language-picker-container" class="container col-xs-12">
+  <ui-select id="language-picker" ng-model="vm.language.selected" theme="bootstrap"
+             ng-change="vm.pickLanguage(vm.language.selected.identifier)" enable-search="false"
```
Search appears to be enabled in the screenshots? Maybe we can clever about this and only enable it when there's more than something like 10 options?

lukeferguson commented on src/core/templates/i18n/language_picker.html:
```diff
@@ -1,7 +1,10 @@
+<div id="language-picker-container" class="container col-xs-12">
+  <ui-select id="language-picker" ng-model="vm.language.selected" theme="bootstrap"
+             ng-change="vm.pickLanguage(vm.language.selected.identifier)" enable-search="false"
+             class="pull-right col-xs-12 col-sm-3 col-md-2 col-lg-2">
```
Enforcing styling is potentially a bit too prescriptive - does this work for studio too? might be better to wrap the component with this instead

lukeferguson commented on src/core/templates/i18n/language_picker.html:
```diff
@@ -1,7 +1,10 @@
-  </ui-select-choices>
-</ui-select>
\ No newline at end of file
+<div id="language-picker-container" class="container col-xs-12">
```
container classes should only ever appear once in the document

TolgaCura commented:
Updated PR
- removed all styling
- Search enabled when 10+ languages
- enabled placeholder translation (displays 'Select...' translation key if no translation present)

TolgaCura commented:
![screen shot 2016-12-22 at 11 35 57](https://cloud.githubusercontent.com/assets/19417743/21424614/da363d66-c83b-11e6-825c-0ee08ea3f0f6.png)
![screen shot 2016-12-22 at 11 36 30](https://cloud.githubusercontent.com/assets/19417743/21424615/da373b6c-c83b-11e6-8419-eafb81350b2d.png)
![screen shot 2016-12-22 at 11 37 02](https://cloud.githubusercontent.com/assets/19417743/21424612/da36328a-c83b-11e6-859a-b2580b34ad64.png)
![screen shot 2016-12-22 at 11 37 16](https://cloud.githubusercontent.com/assets/19417743/21424611/da3621a0-c83b-11e6-9b43-d6af07cd2dc4.png)
![screen shot 2016-12-22 at 11 38 08](https://cloud.githubusercontent.com/assets/19417743/21424613/da3658d2-c83b-11e6-8b4f-566426b04cd3.png)
